concord-consortium/teaching-teamwork
# Added editor for activity JSON

I've added an online editor for the activity JSON.  The editor can save the JSON in the browser's local storage and the teaching-teamwork app has been updated so that activity name's prefixed with 'local:' load from local storage instead of the file system.

You can see it in operation here without pulling down the code:

http://dougmartin.github.io/concord/teaching-teamwork/add-editor/editor/

This wasn't in the SOW but I thought that it could save everyone (especially me) a lot of time generating JSON when testing new components or debugging existing components.  The editor has a built in "linter" so when you have invalid JSON it will show a red X on the left hand gutter of the editor.  

To test it out:

1. Open the link above
2. Either create a new activity or paste in the activity from the activities folder in the code
3. Click the 'Save' button to save the activity (if the save button isn't enabled after the paste enter a space in the editor - sometimes pastes don't register as a change)
4. Click the 'Use' button to open the activity in teaching-teamwork in a new tab 

Anything you 'save' will only be saved in your browser - others won't be able to see it.  If you clear your browser history the saved files will also be cleared.  Until then you can use the 'Open' button which will list the saved files - just double click on a filename to open it.
